### ISSUE TITLE
Scriptlet don't work randomly in Firefox

### ISSUE THREAD

**Issue by ghcsd** [uBlockOrigin/uBlock-issues#1694]
### Prerequisites

- [X] I verified that this is not a filter issue (MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues))
- [X] This is not a [support issue or a question](https://old.reddit.com/r/uBlockOrigin/)
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues) to avoid opening a duplicate issue
- [X] The issue is not present after wholly disabling uBlock Origin ("uBO") in the browser
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### I tried to reproduce the issue when...

- [X] uBO is the only extension
- [X] uBO with default lists/settings
- [X] using a new, unmodified browser profile

### Description

![2021-08-25](https://user-images.githubusercontent.com/32384764/130738022-555f36ad-ef70-46cd-9d1d-7711ea2f8e7a.gif)

 Working properly in Chrome.

### A specific URL where the issue occurs

https://www.zhangxinxu.com/wordpress/

### Steps to Reproduce

1. Add `www.zhangxinxu.com##+js(set, jsonSlide, null)` to `My filters`.
2. Open `https://www.zhangxinxu.com/wordpress/`
3. Refresh the page many times.
4. Open the Web Console `(Ctrl+Shift+K)`, type `jsonSlide` and press `Enter`.

### Expected behavior

Step 3: The QRcode image on the right side will not appear.
Step 4: The Web Console output is `null`.

### Actual behavior

Step 3: The QR code image on the right will appear randomly after the page is refreshed.
Step 4: The web console randomly outputs `null` or `Array[{...}]`.

### uBlock Origin version

uBlock Origin 1.37.3.16

### Browser name and version

Firefox Nightly 93.0a1

### Operating System and version

Window 10.0.19043

**Comment by uBlock-user** [comment]
`zhangxinxu.com##+js(set, jsonSlide, null)` sometimes works and sometimes doesn't, this is a race condition. Scriptlets are not guaranteed to work where race condition is prevalent, this is not specific to Firefox and will occur on other brand of browsers too. You will need to come up with a different filter.

**Comment by felix-22** [comment]
- why not using CHN list? I've enabled CHN list in uBO and the banner is gone, thanks to this filter:
`zhangxinxu.com##.sldebar_out + *,style[id] + a[class]` but it triggers anti-adblock message

- issues related to race condition: 
https://github.com/uBlockOrigin/uBlock-issues/issues/53
https://github.com/gorhill/uBlock/issues/3357

- also, the slider is in `#shadow-root`, related issues: 
https://github.com/uBlockOrigin/uBlock-issues/issues/803
https://github.com/uBlockOrigin/uBlock-issues/issues/880

**Comment by gorhill** [comment]
It's curious that the scriptlet always works when the page is opened in a new tab, but not when reloading the page.

**Comment by gorhill** [comment]
There was a difference in the way the scriptlets are injected in Firefox compared to Chromium, due to be [unreliable in a distant past](https://github.com/gorhill/uBlock/commit/b4306e3297d7fb16d8b9dd438c494e8472440152), but after I tried again now, it seems that injecting the same way as in Chromium causes the scriptlets to be more reliable. After the change, I can still reproduce the issue here, but less frequently.

I still do not understand why the scriptlet seems to have no effect once in a while -- even with a race condition, `window.jsonSlide` should be trapped and return `null` in the dev console after the page has loaded.

**Comment by uBlock-user** [comment]
@gorhill Is https://github.com/uBlockOrigin/uBlock-issues/issues/21 still relevant in Firefox ?

**Comment by gorhill** [comment]
No, injecting by modifying response data was not a good idea in the end.

[contentScripts.register()](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/contentScripts/register) would be what needs to be investigated in some future.

**Comment by gorhill** [comment]
I investigated further, and this was a race-condition in the sense uBO was bailing out from applying the scriptlet if it detected that the property to override was already set and [not-configurable as per `Object.getOwnPropertyDescriptor()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyDescriptor#description). However I found that if instead of bailing out it simply assigns the constant value to the property, this allows uBO to take it over if it was already set by the page.

**Comment by felix-22** [comment]
@yingziwu could you investigate https://github.com/uBlockOrigin/uBlock-issues/issues/1694#issuecomment-907327052 ?
